### PR TITLE
added graph 1

### PR DESCRIPTION
**Pull request by mboom**
Draws a graph with on the x-axis the number of comments devided into groups: 0-2, 2-4, 5-10, etc. On the y-axis the number of pull-requests.

### CODE REVIEW

File: src/modules/graph1.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
+/* globals define, area */
+define(function () {
+    return {
+        name: 'graph1',
+        title: 'Number of pull-request per number of comments',
+        size: 1,
+        parentSelector: '#bodyrow',
+        body: function () {
```
Comment (borek2) [inline src/modules/graph1.js]: Can you divide this big function in different functions, it is very unclear what everything does

**Comment by DaanvanderValk** [comment]
Looks nice! Stuff to do:
- [ ] Fix the JSHint issues (the build fails because of it).
- [ ] Split each bar (vertically) in to a proportion of open, closed and merged pull request.

File: src/modules/graph1.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                    { 'x': 2, 'y': 5 },
+                    { 'x': 4, 'y': 5 }
+                ],
+                domain = ['0-2', '3-5', '5-10', '<10'],
+                buckets = [];
+            for (var i = 0; i < domain.length; ++i) {
+                buckets.push(0);
+            }
+
+            function map(x) {
```
Comment (borek2) [inline src/modules/graph1.js]: lookout with naming, map is used in the global javascript namespace

File: src/modules/graph1.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                    case (x >= 6 && x <= 10): return 2; break; // 6 - 10
+                    case (x >= 3 && x <= 5): return 1; break; // 3 - 5
+                    case (x >= 0 && x <= 2): return 0; break; //0 - 2
+                    default: return 0; break;
+                }
+            }
+            for (var i = 0; i < sizeData.length; ++i) {
+                var item = sizeData[i];
+                buckets[map(item.x)] += item.y;
+                console.log("x: " + map(item.x));
```
Comment (borek2) [inline src/modules/graph1.js]: remove the debug log statements, cramps up the console :)

File: src/modules/moduleList.js [hunk 4]
```diff
@@ -26,7 +26,8 @@ define([], function () {
     	'modules/pr-size',
     	'modules/average-comment-size-compared',
     	'modules/average-comment-size-yours',
-    	'modules/average-comment-size-total'
+    	'modules/average-comment-size-total',
+        'modules/graph1.js'
```
Comment (borek2) [inline src/modules/moduleList.js]: does jslint likes this little space before the string ? 💃

**Comment by borek2** [comment]
Nice going! Take my comments into consideration, especially about the division of functions, it is very unclear whats happening

**Comment by borek2** [comment]
Oh one other thing, desribe in your PR how your code works, or a summary or something. No description is a nogo (Bastiaan would not like this ;))

File: src/modules/graph1.js (hunk 1 above)
Comment (mboom) [inline src/modules/graph1.js]: Ahmed works on this issue. He moves some code outside the function and puts it into a different function. I will wait until he is finished.

File: src/modules/graph1.js (hunk 2 above)
Comment (mboom) [inline src/modules/graph1.js]: Changed it to devide/1.

File: src/modules/graph1.js (hunk 3 above)
Comment (mboom) [inline src/modules/graph1.js]: All console logs should be removed now.

File: src/modules/moduleList.js (hunk 4 above)
Comment (mboom) [inline src/modules/moduleList.js]: I can't see the little space on my side. Als jslint is not complaining about that.

**Comment by mboom** [comment]
@DaanvanderValk I will split the bars in a next pull-request.

**Comment by agudek** [comment]
Have you also thought about how you want to label the x-axis dynamically according to the received data in the future?

**Comment by mboom** [comment]
@agudek Yes I have thought about that. At this moment you need to maintain the domain variable in line 19 and the devide function/switch statement in line 25 to change the labels on the x-axis. The code takes care of the rest.

**Comment by agudek** [comment]
What I meant was : 
In the future, the number of comment will be retrieved from the DB so will be unknown. These might be between 0 and 10, but might also be >1000. The decision how to label the x-axis should be made after these values are known, but at the moment they are hardcoded in line 19.

**Comment by mboom** [comment]
@agudek I see your point. This will be fixed in a next pull-request.

**Pull request merged**